### ISSUE TITLE
Election code hits EtcdEventIndexCleared when waiting for leader change

### ISSUE THREAD

**Issue by fasaxc** [projectcalico/calico#919]
When not he master, the election code does a watch on the election key to wait for it to change.  However, it doesn't update its waitIndex with each poll so eventually, it'll poll on a cleared index and hit EtcdEventIndexCleared.  This results in a spammy log and a new voting round, which re-reads the key.  I.e. it fixes itself after emitting the log:
```
41616:2015-11-20 14:11:06.714 27706 WARNING calico.election [-] Failed to wait for master change - key /calico/openstack/v1/neutron_election: EtcdEventIndexCleared('The event in requested index is outdated and cleared : the requested history has been cleared [39360/39341]',)
```

**Comment by fasaxc** [comment]
In review on gerrit: https://review.openstack.org/#/c/251892/

**Comment by fasaxc** [comment]
Would be resolved by etcd v3 API #952

**Comment by fasaxc** [comment]
Fixed in networking-calico repo.
